storm-core/src/jvm/org/apache/storm/daemon/StormCommon.java

```diff
@@ -436,7 +414,9 @@ public static void addMetricComponents(Map conf, StormTopology topology) {
     public static void addSystemComponents(Map conf, StormTopology topology) {
         Map<String, StreamInfo> outputStreams = new HashMap<String, StreamInfo>();
         outputStreams.put(Constants.SYSTEM_TICK_STREAM_ID, Thrift.outputFields(Arrays.asList("rate_secs")));
+        outputStreams.put(Constants.METRICS_STREAM_ID, Thrift.directOutputFields(Arrays.asList("task-info", "data-points")));
```

Comment:
Oh sorry I've confused. We send the metrics tuple from each task with fake source stream which is `SYSTEM_TASK_ID METRICS_STREAM_ID`, so it's needed. Please see the change of `metric-tick`.